pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -108,3 +114,19 @@ func (r *Reconciler) ReconcileCNI(ctx context.Context, log logr.Logger, clusterS
 
 	return r.cniReconciler.Reconcile(ctx, log, client, clusterSpec)
 }
+
+func toClientControlPlane(cp *tinkerbell.ControlPlane) *clusters.ControlPlane {
+	other := make([]client.Object, 0, 1)
+	if cp.Secrets != nil {
```

Comment:
Are there scenarios where we don't set this Secret, causing it to equal `nil`?

Reply:
When I run the ReconcileControlPlane test, the secret is nil without error, so I assumed it could be. Would you happen to know if this is field is required or not?

Reply:
After some investigation, I can confirm it can be `nil`.registryAuth is not set for the template [here](https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/config/template-cp.yaml#L409
)